src/omotes_simulator_core/adapter/transforms/mappers.py

```diff
@@ -193,7 +193,7 @@ def _get_junction(self) -> dict[str, list[tuple[str, str]]]:
             # Esdl Junctions need to be skipped for now, are added later.
             temp_list = [
                 self.esdl_object.get_connected_assets(
-                    asset_id=esdl_joint.esdl_asset.id, port_id=port
+                    asset_id=esdl_joint.esdl_asset.id, port_id=port[0]
```

Comment:
Why do you need to call the port list with port[0]? Isn't the port already a seperate Port (dataclass) so should you not use port.port_id?

Reply:
The [0] should be removed, now you only get the first character of the port_id string. 
The port dataclass is only used in the mapper. The get_port_ids return a list of strings with the sorted port ids.